web/packages/teleport/src/UnifiedResources/Resources.tsx

```diff
@@ -77,7 +84,7 @@ export function Resources() {
       )}
       <ResourcesContainer gap={2}>
         {fetchedData.agents.map((agent, i) => (
-          <ResourceCard key={i} resource={agent} />
+          <ResourceCard key={i} resource={agent} onLabelClick={onLabelClick} />
```

Comment:
Because these resources will change order a lot (with sorting/filtering), we can't use the index as the key. Since `UnifiedResource` is a type union between all our agents, we'll need to settle on a common "key" that they all share in their types, uniquely. Unforunately, that doesn't exist current. Half of them have a `name` and the other half have an `id`. I'd be fine with just adding one of those to the missing agents and crafting it however we need to, avoiding duplication by maybe appending `kind` to it? something like `name/kind` or `id/kind`.


We can also think about doing this in the backend if it makes more sense there. You let me know

Reply:
You're right, that's not how it should be done. That's the reason why I launched an email thread once about unifying resource IDs, but I think we didn't reach any conclusion. It can be a UUID, a `kind/name` pair, or whatever; from the client side perspective, it should be an opaque string, and we shouldn't even attempt to parse it. All the information that could potentially be derived, like the resource kind, should be passed in a structured form anyway; this way we allow ourselves some degree of freedom and we won't need to hunt all the pieces of code that manipulate strings on the client side to potentially fetch useful information (and half the time doing it wrong). And yes, it means it should be the server's responsibility to generate this. I'll take care of it in a separate PR, if you don't mind.

Reply:
Added an issue for this: https://github.com/gravitational/teleport/issues/30259